core/services/relay/evm/method_binding.go

```diff
@@ -79,14 +85,41 @@ func (m *methodBinding) GetLatestValue(ctx context.Context, params, returnValue
 		Data: data,
 	}
 
-	bytes, err := m.client.CallContract(ctx, callMsg, nil)
+	block, err := m.blockNumberFromConfidence(ctx, confidenceLevel)
+	if err != nil {
+		return err
+	}
+
+	bytes, err := m.client.CallContract(ctx, callMsg, block)
 	if err != nil {
 		return fmt.Errorf("%w: %w", commontypes.ErrInternal, err)
 	}
 
-	return m.codec.Decode(ctx, bytes, returnValue, wrapItemType(m.contractName, m.method, false))
+	return m.codec.Decode(ctx, bytes, returnVal, wrapItemType(m.contractName, m.method, false))
 }
 
 func (m *methodBinding) QueryKey(_ context.Context, _ query.KeyFilter, _ query.LimitAndSort, _ any) ([]commontypes.Sequence, error) {
 	return nil, nil
 }
+
+func (m *methodBinding) blockNumberFromConfidence(ctx context.Context, confidenceLevel primitives.ConfidenceLevel) (*big.Int, error) {
+	value, ok := m.confirmationsMapping[confidenceLevel]
+	if !ok {
+		// TODO is this ok? Maybe some things have to always be faster than Finalized?
```

Comment:
TODO for this PR or follow-up?

Reply:
Resolved this, all or 99%(in case that I missed some) use cases always call contracts with nil block passed into `CallContract()` or nil block in` bind.CallOpts{}` if geth wrapper is used. So I added an error log and fall back to nil block here

Reply:
Resolved this, almost all calls to `CallContract()` or  to geth wrappers with `bind.CallOpts{}` use nil value for blockNumber. Makes sense to default to this with an error log. Exception being if the caller asked for Finalized data where it may be dangerous to return non Finalized state, here it makes sense to throw an error.